jerry-core/include/jerryscript-core.h

```diff
@@ -666,6 +670,19 @@ typedef enum
 } jerry_typedarray_type_t;
 
 
+/**
+ * Container types.
+ */
+typedef enum
+{
+  JERRY_CONTAINER_TYPE_MAP = 0, /**< Map type */
+  JERRY_CONTAINER_TYPE_SET, /**< Set type */
+  JERRY_CONTAINER_TYPE_WEAKMAP, /**< WeakMap type */
+  JERRY_CONTAINER_TYPE_WEAKSET, /**< WeakSet type */
+  JERRY_CONTAINER_TYPE_INVALID, /**< Invalid container */
```

Comment:
It would be better to have the INVALID as the first element (just like in case of typedarrays). That way one can do something like this:
```c
jerry_container_type_t type = jerry_get_container_type (value);
if (!type)
{
   // do some error handling
}
```

Having the invalid elements as a value of 0 would make it consistent. Of course it is best to write out the _INVALID enum value in the check.